src/uct/sm/mm/base/mm_ep.c

```diff
@@ -86,23 +86,25 @@ uct_mm_ep_get_remote_seg(uct_mm_ep_t *ep, uct_mm_seg_id_t seg_id, size_t length,
     return uct_mm_ep_attach_remote_seg(ep, seg_id, length, address_p);
 }
 
-
 /* send a signal to remote interface using Unix-domain socket */
 static void uct_mm_ep_signal_remote(uct_mm_ep_t *ep)
 {
     uct_mm_iface_t *iface = ucs_derived_of(ep->super.super.iface, uct_mm_iface_t);
     char dummy = 0;
     int ret;
 
+    ucs_trace("ep %p: signal remote", ep);
+
     for (;;) {
         ret = sendto(iface->signal_fd, &dummy, sizeof(dummy), 0,
-                     (const struct sockaddr*)&ep->signal.sockaddr,
-                     ep->signal.addrlen);
+                     (const struct sockaddr*)&ep->fifo_ctl->signal_sockaddr,
```

Comment:
do you think that it makes sense?
I guess it could additional overhead to get sockaddr from a shared region of memory.
or the idea that syscall adds more overhead than touching memory, right?
IMO this can vary, and better to not rely on this

Reply:
cost of reading shared variable is negligible compared to syscall and interrupt to remote process to wake it, especially that we already read "head" so it's probably already in sender's cache.

Reply:
on other hand, removing it from ep structure can reduce cache misses due to smaller memory footprint and simplify the code

Reply:
> especially that we've just read "head" so it's probably already in sender's cache.

ah, makes sense then

Reply:
> on other hand, removing it from ep structure can reduce cache misses due to smaller memory footprint and simplify the code

I see, good catch!